### PR TITLE
feat: add BaseLogitsProcessor core interface

### PR DESCRIPTION
**Pull request by bhuvan002**
#### Overview:

This PR adds a base logits processor class to use with different dynamo backends.

#### Details:

This PR adds a base logits processor class that is backend (vllm/trtllm) agnostic and can be easily extended to implement custom logit processing logic. Additional backend specific utilities and wrappers will be added in later PRs.

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)

- closes GitHub issue: #xxx

## Summary by CodeRabbit

* **New Features**
  * Added a public BaseLogitsProcessor interface for logits processing, enabling users to implement and plug in custom processors across supported backends.
  * Simplified imports by exposing the interface at the package level for direct access.

### CODE REVIEW

**Comment by grahamking** [comment]
Looks reasonable to me.

**Comment by bhuvan002** [comment]
> Looks reasonable to me.

Thank you. I was going to add the TRT-LLM plumbing to this PR too, but I will just merge this first and add backend specific things in follow up PRs in the interest of keeping PRs small.

**Review by grahamking — APPROVED** [review]
